omod/src/main/java/org/openmrs/module/openconceptlab/web/rest/resources/ImportResource.java

```diff
@@ -173,4 +178,27 @@ private UpdateScheduler getUpdateScheduler() {
     private static ImportService getImportService() {
         return Context.getService(ImportService.class);
     }
+
+    @Override
+    public Object upload(MultipartFile multipartFile, RequestContext requestContext) throws ResponseException, IOException {
+
+        ImportService importService = getImportService();
+
+        Importer importer = Context.getRegisteredComponent("openconceptlab.importer", Importer.class);
+        importer.setImportService(importService);
+
+        File file = new File(multipartFile.getOriginalFilename());
+        multipartFile.transferTo(file);
+        ZipFile zipFile = new ZipFile(file);
+
+        importer.run(zipFile);
```

Comment:
@tmarzeion, are you working on this? My suggestion would be to actually store the file in memory in ByteArrayInputStream (MultipartFile is discarded after request is compete). You could add updateScheduler.scheduleNow(InputStream inputStream), which would create a new Import object and add a reference to point to that stream on the import object. Configure importer to use that Import object by adding Importer.setImport(Import anImport) (note you will have to adjust runAndHandleErrors to use that object when calling startImport).